community/assessments/projects/openebs/self-assessment.md

```diff
@@ -0,0 +1,297 @@
+- **Local PV LVM Node plugin:** A CSI-node plugin which uses in-built LVM RPC client to carry out volume provisioning, de-provisioning, expansion, snapshot creation for local ZFS volumes. It mounts hostpath directories on cluster hosts to enable communication with LVM kernel modules and block device nodes.
+- **Replicated PV Mayastor REST API:** This exposes a storage management API which may be used directly or via the bundled CSI plugin achitecture to perform CRUD-like operations on logical storage resources.
+- **Replicated PV Mayastor Core Agent:** This is acts as a control-plane for a Mayastor cluster. Communitcates with other mayastor services via HTTP (gRPC).
+- **Replicated PV Mayastor Etcd persistent store:** This persists the state of a Mayastor cluster. Uses replication and self-healing for redundancy and high-availability.
+- **Replicated PV Mayastor HA Cluster Agent:** This is a Mayastor control-plane agent which provides highly available volume target management. This communicates to the Mayastor's core agent via HTTP (gRPC).
+- **Replicated PV Mayastor HA Node Agent:** This is a Mayastor control-plane agent which mounts a hostpath directory and makes use of NVMe commands to perform these actions w.r.t volume target failovers -- detection, reporting and replacement of failed target paths.
+- **Replicated PV Mayastor CSI Controller plugin:** This is a CSI-controller plugin which communicates with the Mayastor storage API (HTTP) and the Kubernetes APIs to orchestrate volume provisioning, de-provisioning, expansion, snapshot operations for Mayastor volumes
+- **Replicated PV Mayastor CSI Node plugin:** This is a CSI-node plugin which communicates with the Mayastor control-plane via HTTP (gRPC) and executes host-level volumes operations. It mounts hostpath directories for accessing sysfs APIs.
+- **Replicated PV Mayastor IO Engine:** This is a userspace storage controller which polls for IO requests and serves volume targets for Kubernetes containers. It consumes a high degree of CPU and memory resources to provide low-lantency, resilient storage. This communicates with the Mayastor control plane using HTTP (gRPC).
+- **Replicated PV Mayastor IO Engine metrics exporter:** This exposes volume controller stats data and pool usage data in prometheus-compatible format. This communicates with IO engines using intra Pod IPC.
+- **Replicated PV Mayastor Stats and Call-home plugin:** This is a plugin for reporting anonymous usage data from the Kubernetes cluster. It communicates with the Kubernetes API, and the Mayastor storage API to collect data.
+- **Clients:** This actor interacts with an OpenEBS cluster using standard Kubernetes tools and/or specialised clients for accessing storage layer functionality. This is usually a Kubernetes cluster admin or a storage admin.
+
+### Actions
+
+#### PVC-PV Based Volume Operations
+
```

Comment:
This probably would need to be more detailed for a joint assessment, but right now I think it's fine.

Reply:
Okay.